### ISSUE TITLE
Linux Issue Tracking

### ISSUE THREAD

**Issue by the-m4a** [civfanatics/CQUI_Community-Edition#203]
I wanted to create a new thread to track the things I saw in Linux.  I have managed to get the game started in a Windows Subsystem for Linux environment with the Live Tuner attached.
It's very unlikely I'd be able to play an entire game in this environment - it's *insanely* slow, but it allows me to load up the game and start it and see what's going on.

My start point is using the changes from RustyD, with the modellenspanel.xml and lua files also renamed to the lower-cased version.

Summary of Current Status:
- Still not seeing the minimap function in Exp2 game
- I can see the banner when loading a base game (no expansions, CQUI is only mod loaded)!

**Comment by the-m4a** [comment]
**Starting a new game**
Mods: Exp1 and Exp2, CQUI, and Better Civ Colors only
During game load:
```
 Runtime Error: C:\Emu\AppDataParent\Sid Meier's Civilization VI\Mods\CQUI\Integrations\ML\Lenses\ModLens_Builder_Config_Default.lua:411: attempt to index a nil value
stack traceback:
	C:\Emu\AppDataParent\Sid Meier's Civilization VI\Mods\CQUI\Integrations\ML\Lenses\ModLens_Builder_Config_Default.lua:411: in function '(main chunk)'
	[C]: in function 'lInclude'
	C:\Emu\AppDataParent\Sid Meier's Civilization VI\Mods\CQUI\Integrations\ML\UI\minimappanel.lua:9: in function '(main chunk)'
	[C]: in function 'lInclude'
	C:\Emu\AppAssets\DLC\expansion2\UI\Replacements\MinimapPanel_Expansion1.lua:6: in function '(main chunk)'
	[C]: in function 'lInclude'
	C:\Emu\AppAssets\DLC\expansion2\UI\Replacements\MinimapPanel_Expansion2.lua:6: in function '(main chunk)'
	[C]: in function '(anonymous)'
Lua callstack:

@C:\Emu\AppDataParent\Sid Meier's Civilization VI\Mods\CQUI\Integrations\ML\UI\minimappanel.lua:9
@C:\Emu\AppAssets\DLC\expansion2\UI\Replacements\MinimapPanel_Expansion1.lua:6
@C:\Emu\AppAssets\DLC\expansion2\UI\Replacements\MinimapPanel_Expansion2.lua:6
```

Another error:
```
 MinimapPanel: InitLens - Adding ModLens: ML_BARBARIAN
 Runtime Error: C:\Emu\AppAssets\DLC\expansion2\UI\Replacements\MinimapPanel_Expansion2.lua:146: attempt to index a nil value
stack traceback:
	C:\Emu\AppAssets\DLC\expansion2\UI\Replacements\MinimapPanel_Expansion2.lua:146: in function 'LateInitialize'
	C:\Emu\AppDataParent\Sid Meier's Civilization VI\Mods\CQUI\Integrations\ML\UI\minimappanel.lua:1541: in function 'OnInit'
Lua callstack:
 TradeOverview: Initializing BTS Trade Overview
```

Finally:
```
 Runtime Error: C:\Emu\AppDataParent\Sid Meier's Civilization VI\Mods\CQUI\Integrations\ML\Lenses\ModLens_Builder_Config_Default.lua:411: attempt to index a nil value
stack traceback:
	C:\Emu\AppDataParent\Sid Meier's Civilization VI\Mods\CQUI\Integrations\ML\Lenses\ModLens_Builder_Config_Default.lua:411: in function '(main chunk)'
	[C]: in function 'lInclude'
	C:\Emu\AppDataParent\Sid Meier's Civilization VI\Mods\CQUI\Integrations\ML\UI\Panels\modallenspanel.lua:12: in function '(main chunk)'
	[C]: in function 'lInclude'
	C:\Emu\AppAssets\DLC\expansion2\UI\Replacements\ModalLensPanel_Expansion1.lua:6: in function '(main chunk)'
	[C]: in function 'lInclude'
	C:\Emu\AppAssets\DLC\expansion2\UI\Replacements\ModalLensPanel_Expansion2.lua:6: in function '(main chunk)'
	[C]: in function '(anonymous)'
```

**Comment by the-m4a** [comment]
When founding a city - presumably the issue with the city banners not showing in the Expansions:
```
 Runtime Error: C:\Emu\AppAssets\Base\Assets\ui\instancemanager.lua:179: attempt to index a nil value
stack traceback:
	C:\Emu\AppAssets\Base\Assets\ui\instancemanager.lua:179: in function 'BuildInstance'
	C:\Emu\AppAssets\Base\Assets\ui\instancemanager.lua:36: in function 'GetInstance'
	C:\Emu\AppAssets\DLC\expansion2\UI\CityBanners\citybannermanager.lua:249: in function 'CityBanner:Initialize'
	C:\Emu\AppDataParent\Sid Meier's Civilization VI\Mods\CQUI\Assets\UI\WorldView\citybannermanager_CQUI.lua:115: in function 'CQUI_Common_CityBanner_Initialize'
	C:\Emu\AppDataParent\Sid Meier's Civilization VI\Mods\CQUI\Assets\UI\WorldView\citybannermanager_CQUI_expansions.lua:29: in function 'CityBanner.Initialize'
	C:\Emu\AppAssets\DLC\expansion2\UI\CityBanners\citybannermanager.lua:157: in function 'CityBanner:new'
	(tail call): ?
	C:\Emu\AppAssets\DLC\expansion2\UI\CityBanners\citybannermanager.lua:2606: in function 'OnCityAddedToMap'
	[C]: in function 'func'
	[C]: in function '(anonymous)'
 MapLabelManager: PANTHALASSIC OCEAN (1228.7135009766, 333.06298828125)
```

**Comment by Infixo** [comment]
All of the above from the CQUI-master branch?

**Comment by the-m4a** [comment]
Yes - or at least what my VSCode Git was claiming to be master, plus two sets of changes:
1. The changes currently in RustyD's PR (where all CQUI mentions of minimappanel.lua and minimappanel.xml are lower-cased), in the file system and in the modinfo file
2. All CQUI mentions of modallenspanel.lua and and modallenspanel.xml are lower cased

#2 was present already in my version of master branch, but I need to verify that this was somehow not a hold-over from the branch where I was updating the ModLens integration.  I didn't get the opportunity to take a look last night, and I'm off work next week so I'll have more time to dig back in.

Also, I ended up looking into the instancemanager issue, as it's very confusing in terms of what is happening there.  I neglected to paste the line above the error message, but it's that "invalid root element" error.  The stuff that CQUI does in that stack above is minimal, citybannermanager_CQUI just immediately calls the base version of CityBanner:Initialize, for example.  citybannermanager_CQUI_expansions doesn't make any changes to the objects passed  in the parameters, either... so, it's strange, or a situation where this error is popping as the result of something else breaking, but isn't showing up in the Live Tuner.

Also for what it's worth, I did confirm that there is no Lua.log file showing up.

**Comment by Infixo** [comment]
Can you start a game with the master branch and post lua.log?

**Comment by the-m4a** [comment]
Here is the lua log from master branch.
Expansion 2, Duel sized map, Standard game, CQUI is the only not-Firaxis mod loaded.
```
Working late?
[ Lua State = Main State ]
 AdvancedSetup: Player Count Changed
 AdvancedSetup: There are 2 participating players.
 JoiningRoom: OnFinishedGameplayContentConfigure() g_waitingForContentConfigure=true
 AdvancedSetup: Hiding Game Setup
 LoadScreen: true
 LoadScreen: true
 LoadScreen: false
 Map Script: Generating Continents Map
 Map Script: Generating Plot Types
 Map Script: Existing Ratio	21.625
 Map Script: New Mountains 	4
 Map Script: Mountain Set	5
 Map Script: Generating Terrain Types
 Map Script: -
 Map Script: - Desert Percentage:	25
 Map Script: --- Latitude Readout ---
 Map Script: - All Grass End Latitude:	0.1
 Map Script: - Desert Start Latitude:	0.2
 Map Script: - Desert End Latitude:	0.5
 Map Script: - Tundra Start Latitude:	0.65
 Map Script: - Snow Start Latitude:	0.8
 Map Script: - - - - - - - - - - - - - -
 Map Script: Expanding coasts
 Map Script: After Adding Hills: 	198
 Map Script: Desired Volcanoes: 1.384
 Map Script: Continent Boundary Plots: 0
 Map Script: Boundary Plots Per Volcano: 0
 Map Script: Continent Edge Volcanoes Placed: 0
 Map Script: Volcano Placed at (x, y): 41, 17
 Map Script: Volcano Placed at (x, y): 6, 6
 Map Script: Total Volcanoes Placed: 2
 Map Script: Map Generation - Adding Rivers
 Map Script: Map Generation - Adding Lakes
 Map Script: 3 lakes added
 Map Script: Adding Features
 Map Script: Permanent Ice Tiles: 25.74
 Map Script: iPhaseIndex: 7, iIceTilesToAdd: 25.74, RandomEventEnum: 28
 Map Script: iPhaseIndex: 6, iIceTilesToAdd: 25.74, RandomEventEnum: 27
 Map Script: iPhaseIndex: 5, iIceTilesToAdd: 25.74, RandomEventEnum: 26
 Map Script: iPhaseIndex: 4, iIceTilesToAdd: 17.16, RandomEventEnum: 25
 Map Script: iPhaseIndex: 3, iIceTilesToAdd: 17.16, RandomEventEnum: 24
 Map Script: iPhaseIndex: 2, iIceTilesToAdd: 17.16, RandomEventEnum: 23
 Map Script: iPhaseIndex: 1, iIceTilesToAdd: 17.16, RandomEventEnum: 22
 Map Script: Number of Tiles: 	297
 Map Script: Number of Forests: 	54
 Map Script: Percent Forests: 	18.181818181818
 Map Script: Number of Jungles: 	31
 Map Script: Percent Jungles: 	10.43771043771
 Map Script: Percent of Junglable: 	40.25974025974
 Map Script: Number of Marshes: 	9
 Map Script: Adding cliffs
 Map Script: In NaturalWonderGenerator.Create()
 Map Script:     Placing 2 Natural Wonders
 Map Script: Feature Type: 6, Valid Hexes: 58
 Map Script: Feature Type: 7, Valid Hexes: 1
 Map Script: Feature Type: 8, Valid Hexes: 7
 Map Script: Feature Type: 9, Valid Hexes: 2
 Map Script: Feature Type: 10, Valid Hexes: 0
 Map Script: Feature Type: 11, Valid Hexes: 31
 Map Script: Feature Type: 12, Valid Hexes: 3
 Map Script: Feature Type: 13, Valid Hexes: 2
 Map Script: Feature Type: 14, Valid Hexes: 2
 Map Script: Feature Type: 15, Valid Hexes: 6
 Map Script: Feature Type: 16, Valid Hexes: 30
 Map Script: Feature Type: 17, Valid Hexes: 1
 Map Script: Feature Type: 18, Valid Hexes: 8
 Map Script: Feature Type: 19, Valid Hexes: 4
 Map Script: Feature Type: 20, Valid Hexes: 9
 Map Script: Feature Type: 21, Valid Hexes: 5
 Map Script: Feature Type: 22, Valid Hexes: 2
 Map Script: Feature Type: 23, Valid Hexes: 0
 Map Script: Feature Type: 24, Valid Hexes: 5
 Map Script: Feature Type: 31, Valid Hexes: 8
 Map Script: Feature Type: 32, Valid Hexes: 42
 Map Script: Feature Type: 33, Valid Hexes: 4
 Map Script: Feature Type: 34, Valid Hexes: 8
 Map Script: Feature Type: 35, Valid Hexes: 32
 Map Script: Feature Type: 36, Valid Hexes: 6
 Map Script: Feature Type: 37, Valid Hexes: 2
 Map Script: Feature Type: 38, Valid Hexes: 44
 Map Script: Feature Type: 39, Valid Hexes: 36
 Map Script: Feature Type: 40, Valid Hexes: 6
 Map Script: Num wonders with valid location: 27
 Map Script:  Selected Wonder = 14, Random Score = 	92
 Map Script:  Set Wonder with Feature ID of 20 at location (41, 21)
 Map Script:  Selected Wonder = 3, Random Score = 	89
 Map Script:  Selected Wonder = 25, Random Score = 	87
 Map Script:  Selected Wonder = 27, Random Score = 	87
 Map Script:  Set Wonder with Feature ID of 39 at location (29, 18)
 Map Script: Adding Features from Continents
 Map Script: Full-Map Fissure Placed at (x, y): 37, 20
 Map Script: Full-Map Fissure Placed at (x, y): 34, 16
 Map Script: Number of Oasis: 	0
 Map Script: Number of Fissures: 	2
 Map Script: Map Generation - Marking Coastal Lowlands
 Map Script: 42 Coastal Lowland tiles added
 Map Script:   45% of eligible coastal tiles
 Map Script: In ResourceGenerator.Create()
 Map Script:     Placing resources
 Map Script: Creating start plot database.
 Map Script: Major Start X: 	35	Major Start Y: 	9
 Map Script: Major Start X: 	15	Major Start Y: 	14
 Map Script: Minor Start X: 	40	Minor Start Y: 	17
 Map Script: Minor Start X: 	31	Minor Start Y: 	15
 Map Script: Minor Start X: 	8	Minor Start Y: 	18
 Map Script: -------------------------------
 Map Script: Map Generation - Adding Goodies
 Map Script: -------------------------------
 cqui_main: CQUI Loaded! Update for v1.0.3.31 (2020-07-24)
 WorldCongress: Initializing World Congress Lua
 MapLabelManager: ATLANTIC OCEAN (455.83233642578, -367.03909301758)
 MapLabelManager: ATLANTIC OCEAN (513.69268798828, 421.98709106445)
 MapLabelManager: ATLANTIC OCEAN (-139.48513793945, 416.24395751953)
 MapLabelManager: ATLANTIC OCEAN (-1021.7616577148, 518.18298339844)
 MapLabelManager: ATLANTIC OCEAN (-686.90802001953, -362.15008544922)
 MapLabelManager: ATLANTIC OCEAN (1145.2681884766, -591.43530273438)
 MapLabelManager: ATLANTIC OCEAN (-103.55373382568, -380.02630615234)
 NotificationPanel: NotificationPanel_CQUI.lua: File loaded
 MapSearchPanel: Created SearchContext 'MapSearch_Primary'
 MapSearchPanel: Created SearchContext 'MapSearch_Suggestions'
 MinimapPanel: Governor Builder Index = 4
 MinimapPanel: Governor Builder Aquaculture hash = -122625491
 MinimapPanel: Governor Builder Parks Rec hash = 1630405030
 Runtime Error: C:\Emu\AppDataParent\Sid Meier's Civilization VI\Mods\CQUI\Integrations\ML\Lenses\ModLens_Builder_Config_Default.lua:411: attempt to index a nil value
stack traceback:
	C:\Emu\AppDataParent\Sid Meier's Civilization VI\Mods\CQUI\Integrations\ML\Lenses\ModLens_Builder_Config_Default.lua:411: in function '(main chunk)'
	[C]: in function 'lInclude'
	C:\Emu\AppDataParent\Sid Meier's Civilization VI\Mods\CQUI\Integrations\ML\UI\minimappanel.lua:9: in function '(main chunk)'
	[C]: in function 'lInclude'
	C:\Emu\AppAssets\DLC\expansion2\UI\Replacements\MinimapPanel_Expansion1.lua:6: in function '(main chunk)'
	[C]: in function 'lInclude'
	C:\Emu\AppAssets\DLC\expansion2\UI\Replacements\MinimapPanel_Expansion2.lua:6: in function '(main chunk)'
	[C]: in function '(anonymous)'
Lua callstack:

@C:\Emu\AppDataParent\Sid Meier's Civilization VI\Mods\CQUI\Integrations\ML\UI\minimappanel.lua:9
@C:\Emu\AppAssets\DLC\expansion2\UI\Replacements\MinimapPanel_Expansion1.lua:6
@C:\Emu\AppAssets\DLC\expansion2\UI\Replacements\MinimapPanel_Expansion2.lua:6
 MinimapPanel: Governor Builder Index = 4
 MinimapPanel: Governor Builder Aquaculture hash = -122625491
 MinimapPanel: Governor Builder Parks Rec hash = 1630405030
 MinimapPanel: Initializing 7 lenses
 MinimapPanel: InitLens - Adding ModLens: ML_NATURALIST
 MinimapPanel: InitLens - Adding ModLens: ML_WONDER
 MinimapPanel: InitLens - Adding ModLens: ML_SCOUT
 MinimapPanel: InitLens - Adding ModLens: ML_BUILDER
 MinimapPanel: InitLens - Adding ModLens: ML_BARBARIAN
 MinimapPanel: InitLens - Adding ModLens: CQUI_CITIZEN_MANAGEMENT
 MinimapPanel: InitLens - Adding ModLens: ML_ARCHAEOLOGIST
 Runtime Error: C:\Emu\AppAssets\DLC\expansion2\UI\Replacements\MinimapPanel_Expansion2.lua:146: attempt to index a nil value
stack traceback:
	C:\Emu\AppAssets\DLC\expansion2\UI\Replacements\MinimapPanel_Expansion2.lua:146: in function 'LateInitialize'
	C:\Emu\AppDataParent\Sid Meier's Civilization VI\Mods\CQUI\Integrations\ML\UI\minimappanel.lua:1541: in function 'OnInit'
Lua callstack:
 TradeOverview: Initializing BTS Trade Support Tracker
 TradeRouteChooser: Initializing BTS Trade Route Chooser
 TradeRouteChooser: Initializing BTS Trade Support Automater
 TradeOriginChooser: Initializing BTS Trade Origin Chooser
 ModalLensPanel: Governor Builder Index = 4
 ModalLensPanel: Governor Builder Aquaculture hash = -122625491
 ModalLensPanel: Governor Builder Parks Rec hash = 1630405030
 Runtime Error: C:\Emu\AppDataParent\Sid Meier's Civilization VI\Mods\CQUI\Integrations\ML\Lenses\ModLens_Builder_Config_Default.lua:411: attempt to index a nil value
stack traceback:
	C:\Emu\AppDataParent\Sid Meier's Civilization VI\Mods\CQUI\Integrations\ML\Lenses\ModLens_Builder_Config_Default.lua:411: in function '(main chunk)'
	[C]: in function 'lInclude'
	C:\Emu\AppDataParent\Sid Meier's Civilization VI\Mods\CQUI\Integrations\ML\UI\Panels\modallenspanel.lua:12: in function '(main chunk)'
	[C]: in function 'lInclude'
	C:\Emu\AppAssets\DLC\expansion2\UI\Replacements\ModalLensPanel_Expansion1.lua:6: in function '(main chunk)'
	[C]: in function 'lInclude'
	C:\Emu\AppAssets\DLC\expansion2\UI\Replacements\ModalLensPanel_Expansion2.lua:6: in function '(main chunk)'
	[C]: in function '(anonymous)'
Lua callstack:

@C:\Emu\AppDataParent\Sid Meier's Civilization VI\Mods\CQUI\Integrations\ML\UI\Panels\modallenspanel.lua:12
@C:\Emu\AppAssets\DLC\expansion2\UI\Replacements\ModalLensPanel_Expansion1.lua:6
@C:\Emu\AppAssets\DLC\expansion2\UI\Replacements\ModalLensPanel_Expansion2.lua:6
 ModalLensPanel: Governor Builder Index = 4
 ModalLensPanel: Governor Builder Aquaculture hash = -122625491
 ModalLensPanel: Governor Builder Parks Rec hash = 1630405030
 MapPinPopup: initializing MapTacks.IconOptions(0)
 MapPinPopup: Selected 16 columns
 DiplomacyDealView: CQUI Diplomacy Deal View for expansion 2 loaded
 ModLens_CityOverlap: ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 ModLens_CityOverlap:           City Overlap Panel
 ModLens_CityOverlap: ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 ModLens_Resource: ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 ModLens_Resource:            Resource Panel
 ModLens_Resource: ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 TutorialUIRoot: Tutorial: Firaxis in game tutorial prompts.
 TutorialUIRoot: Version: 1
 TutorialUIRoot: Loading bank of items for tutorial scenario: 'BASE'
 MapLabelManager: ATLANTIC OCEAN (455.83233642578, -367.03909301758)
 MapLabelManager: ATLANTIC OCEAN (513.69268798828, 421.98709106445)
 MapLabelManager: ATLANTIC OCEAN (-139.48513793945, 416.24395751953)
 MapLabelManager: ATLANTIC OCEAN (-1021.7616577148, 518.18298339844)
 MapLabelManager: ATLANTIC OCEAN (-686.90802001953, -362.15008544922)
 MapLabelManager: ATLANTIC OCEAN (1145.2681884766, -591.43530273438)
 MapLabelManager: ATLANTIC OCEAN (-103.55373382568, -380.02630615234)
 LoadScreen: OnLoadGameViewStateDone
 MinimapPanel: InitLens - Adding ModLens: ML_CITYOVERLAP
 MinimapPanel: InitLens - Adding ModLens: ML_RESOURCE
 ================ load screen done =================
 ================ Creating new city with Settler =================================
 CityBannerManager: Instance Manager built with bad Root Control [CityBanner] [Anchor]
 Runtime Error: C:\Emu\AppAssets\Base\Assets\ui\instancemanager.lua:179: attempt to index a nil value
stack traceback:
	C:\Emu\AppAssets\Base\Assets\ui\instancemanager.lua:179: in function 'BuildInstance'
	C:\Emu\AppAssets\Base\Assets\ui\instancemanager.lua:36: in function 'GetInstance'
	C:\Emu\AppAssets\DLC\expansion2\UI\CityBanners\citybannermanager.lua:249: in function 'CityBanner:Initialize'
	C:\Emu\AppDataParent\Sid Meier's Civilization VI\Mods\CQUI\Assets\UI\WorldView\citybannermanager_CQUI.lua:115: in function 'CQUI_Common_CityBanner_Initialize'
	C:\Emu\AppDataParent\Sid Meier's Civilization VI\Mods\CQUI\Assets\UI\WorldView\citybannermanager_CQUI_expansions.lua:29: in function 'CityBanner.Initialize'
	C:\Emu\AppAssets\DLC\expansion2\UI\CityBanners\citybannermanager.lua:157: in function 'CityBanner:new'
	(tail call): ?
	C:\Emu\AppAssets\DLC\expansion2\UI\CityBanners\citybannermanager.lua:2606: in function 'OnCityAddedToMap'
	[C]: in function 'func'
	[C]: in function '(anonymous)'
 PlotInfo: PlotInfo::OnCitySelectionChanged	0	65536	35	9	0	true	false
```

**Comment by the-m4a** [comment]
also - for what it's worth:
- the banners actually load and show in the vanilla (no expansions loaded) game

The City panels appear to show okay (if you can get to the city) in expansion 2 game (this is the game from the lua log above)
![image](https://user-images.githubusercontent.com/8787640/99935867-8d6a0c80-2d16-11eb-8bf4-8f3cced93c68.png)

**Comment by stevenschlansker** [comment]
I can confirm that with #220 merged in, Minimap starts working again.
City banners are still broken.
Additionally, I noticed that eventually Game Options gets broken - all text labels disappear.  This doesn't seem to happen immediately on mod load but later (e.g. after starting a game) and persists until you unload the mod and restart Civ.

I looked and looked and looked for a `lua.log` to share but could not find one in my `~/.local/share/aspyr-media/SMCVI/Logs/` although there's many other log files there.

![image](https://user-images.githubusercontent.com/129097/101992956-81190580-3c6b-11eb-94fd-eb0820aeceab.png)

**Comment by stevenschlansker** [comment]
Ah, it turns out the disappearing game options might be a conflict between CQUI and Concise UI which I had also inadvertently enabled.  Sorry for the noise if that's the case.

**Comment by stevenschlansker** [comment]
I figured out a workaround:  if you run ZFS, you can create a case insensitive filesystem: 
```
sudo zfs create -o casesensitivity=insensitive "tank/civ6"
```
If you move the Civ6 install as well as the CQUI mod install to this new filesystem, the city banners appear again.
This leads me to believe the city banners disappearing is another instance of incorrect filename casing.
So far I have not noticed any other issues with this workaround.

**Comment by stevenschlansker** [comment]
Ah, and as soon as I reported "no issues" I found the next one:
Government doesn't work.  clicking on the Government icon at the top left doesn't show any box, and if your turn includes a mandatory "Fill Policy Slot" it seems impossible to complete your turn.

**Comment by the-m4a** [comment]
I published my current working branch -- the main thing there is I added a "start" and "end" print statement on every lua file, just so I can easily see which ones actually load.  I have managed to get the minimap to load again, but not yet banners.  The map and the banners work in vanilla, there's just something about that whole stuff from the expansions that it barfs all over.

The branch is m4a-linuxDec2020
https://github.com/civfanatics/CQUI_Community-Edition/tree/m4a-linuxDec2020

![image](https://user-images.githubusercontent.com/8787640/102006175-e99bcd00-3cd3-11eb-8136-013771f97ed1.png)

**Comment by the-m4a** [comment]
It doesn't seem to be picking up the XML files for the city banners... it'll load the lua files, but it doesn't load the CQUI version of the XML.  and of course no hint as to why in the Lua output or in the Modding.log file.  Weird.

**Comment by the-m4a** [comment]
Some progress!  (Sort of, anyway)...
I was able to confirm the theory that the XML Include directives are failing for some reason, though we are still without an effective way to actually determine what's happening.

![image](https://user-images.githubusercontent.com/8787640/102176733-ea10a100-3e56-11eb-9c14-f5a8a63ff59f.png)

I did the following to get to this point:
1. Added a _citybannermanager.xml_ to the CQUI Expansion2 CityBanners folder
2. Commented out the <Include File="CityBannerInstances"> and <Include File="CityReligionInstances"> lines in _citybannermanager.xml_ and pasted the entirety of _citybannerinstances.xml_ and _cityreligioninstances.xml_ into the _citybannermanager.xml_ file.
3. Reduced _citybannerinstances.xml_ and _cityreligioninstances.xml_ to nearly nothing (each is now a single declaration of a control I added just to see if I could see it as an object in the Live Tuner).
4. Moved just the banner files their own load spot in the modinfo just to assist with troubleshooting this

**So, current status:**
I will push the changes here to GitHub in case any of y'all wants to try and play a few turns and see what you can see.  If you hit an issue please share the savefile, if possible.

Note that this change only currently works for Expansion2, I did not make changes to the Expansion1 files/folders.

I'll keep occasionally poking at this... but perhaps this may be a playable Linux version?  @stevenschlansker @alexeyOnGitHub

**Comment by RustyD** [comment]
I've just run a quick-ish test. Base game seems to be ok as far as minimap, access to CQUI settings, city banners and productions queues.
Expansion 1, there was a minimap and access to CQUI, but no city banners.
Expansion 2, minimap appeared broken, but I did have city banners and productions queues.

**Comment by the-m4a** [comment]
Thanks @RustyD - it appears I forgot to do the file rename (to the correct casing) on the minimappanel.xml files in Integrations\ML\DLC\Expansion1\UI\Replacements and Integrations\ML\DLC\Expansion2\UI\Replacements folders.  I had made the change in the modinfo file and made the change on my local install, but did not do that in the Git branch.  I just made the change in the Git branch, so if you download it again (or just rename those minimappanel.xml files to the all-lowercase) then it should work for EXP2?  

![image](https://user-images.githubusercontent.com/8787640/102293256-90ae7d80-3efb-11eb-80c6-5f99400f1634.png)

**Comment by RustyD** [comment]
@the-m4a I've just done a quick test of EXP2, and the minimap now appears to be working, along with access to the CQUI menu. So, I _think_ everything is good.

**Comment by enderger** [comment]
Here is what I found from minimal research:

#### Base Game
- Everything appears fine

#### R&F 
- City banners are missing

#### Gathering Storm 
- Everything appears fine

#### System
- OS: NixOS Unstable 21.03
- Game: Build 1.0.8.4(212129) (December Patch)

#### Tl;Dr
All that is missing is to apply the patch to Rise&Fall.

**Comment by enderger** [comment]
@stevenschlansker I have experienced that bug with just Concise UI installed, nothing to do with CQUI.

**Comment by the-m4a** [comment]
I pushed an update to the branch.  It includes changes from the current main branch, so there should be no problems with the diplomacy window.

R&F should now work and Base Game and  GS should also continue to be funtional.  Of note: the GS citybanner XML files no longer contain the hack where I manually pull in the citybannerinstance xml and cityreligioninstance xml.  The XML files all do currently contain a container that I added that allows me to verify the file has been loaded as expected...

If GS works and R&F doesn't, then for some reason the CityBannerManager.xml file found in Assets/Expansion1/CityBanners needs to be named with that exact casing.  GS would _not_ work when the citybannermanager.xml file had the "proper" casing... it would only work if the file was entirely lower case.

To sum up:
CQUI Base Game:
- Assets/UI/WorldView/citybannermanager.xml

CQUI R&F:
- Assets/Expansion1/CityBanners/citybannermanager.xml
- Assets/Expansion1/CityBanners/CityBannerInstances.xml
- Assets/Expansion1/CityBanners/CityReligionInstances.xml

CQUI GS:
- Assets/Expansion2/CityBanners/citybannermanager.xml
- Assets/Expansion2/CityBanners/CityBannerInstances.xml
- Assets/Expansion2/CityBanners/CityReligionInstances.xml

Civ 6 Base Game and DLC on my Linux VM:
- all of the XML and LUA file names are lower-case only.  The modinfo files that refer those XML and Lua files do _not_ bother to match the all-lowercase casing of the file names... and as far as I know there's no issues, so go figure.

**Comment by enderger** [comment]
After a full game test with Gathering Storm on NixOS 21.03 unstable, here are the problems I found:
- The notifications appear in a weird place, slightly off the center of the screen
![Notif-Screenshot](https://user-images.githubusercontent.com/36283171/102900519-3e042280-4421-11eb-9349-0ff58552aa0d.png)
- The CQUI options button above the map does not toggle off when you exit the CQUI settings menu.
- When hovering over some bonuses in the government menu, no tooltip appears.

As I play through the game, I'll update this list.

**Comment by the-m4a** [comment]
@enderger Thank you so much, much appreciated!
- **Notifications**:  They do exist in an odd spot now, but that should be the same spot as the unmodified game.  If I remember correct the CQUI ones are a little bit wider and one of them uses a larger font, but the top-left anchor point is either close or what the unmodified game has.
- **CQUI Options**: I'll check into that one
- **Government Menu:** Which bonuses in particular?  I don't think we did anything specific there, so that may not be a CQUI thing.  I will take a peek if there's specific ones that behave that way.
- **Purchasing Tiles:** CQUI has logic that if the mouse moves even a little bit when trying to purchase a tile, it does not complete the purchase in order to guard against accidental clicks.  It's an issue I've wanted to address for a while, basically I need to do something that only cancels it if the mouse moved 5 pixels in any one direction or something.
- **Great Person Text:** Can you get me a screen shot?

**Comment by enderger** [comment]
Happy to help! 

- The notifications are a bit more offset in comparison to the normal ones. They need to move a good bit to the right
- The options button is likely just a problem with the base mod, as it seems to be a very small cosmetic issue
- The tooltips seem to be missing for the first bonuses in Autocracy, Classical Republic, Monarchy, Merchant Republic, Facism, Communism, Synthetic Tecnocracy, and Digital Democracy
- While none of the great people in the game have long enough description text to cause the issue, I'll see if I can find one who has this issue.

**Comment by enderger** [comment]
Regarding the notifications, I actually think that they need to be offset to the right by the width of a notification.

**Comment by enderger** [comment]
![Screenshot_20201222_171836](https://user-images.githubusercontent.com/36283171/102948026-bbf41800-4479-11eb-85ed-d3ae4a9f40d4.png)
Here, the great admeral is the one with the problem.

**Comment by the-m4a** [comment]
I have created a PR (https://github.com/civfanatics/CQUI_Community-Edition/pull/225) that has the changes necessary to get CQUI showing the minimap and the city banners when running on a Linux install.  If you could give that branch a go (m4a-linux-203), I would apprecaite it, as I only played the first turn to verify the banner would appear.

I checked that RF, G&S, and Vanilla all appeared functional.

Also will take a peek at some of the things that @enderger reported, probably after Chistmas, especially if the set of changes in that PR are what's needed!

**Comment by the-m4a** [comment]
Closing this up as Linux works, basically, and I'll open up a new one of these when needed.
